src/msg/async/ProtocolV2.cc

```diff
@@ -2070,10 +2070,11 @@ CtPtr ProtocolV2::send_auth_request(std::vector<uint32_t> &allowed_methods) {
 
   bufferlist bl;
   vector<uint32_t> preferred_modes;
+  auto am = auth_meta;
```

Comment:
I don't think I understand how taking these per-thread shared pointer references help to deal with pipe replacement races. It's not like we're deleting the existing pipe or its reference until this thread is done, or locking the auth_meta to prevent modification...

Reply:
The to-be-replaced connection's auth_meta points to a struct that we're passing in/using here, without the con lock.  If we replace, the new connectin's auth_meta over and the existing connection's auth_meta pointer is updated.  This keeps the old stale auth_meta struct alive whlie that happens.

Reply:
Ooooh I see it now, that works.